File: backend/app/schemas/google_books.py
from pydantic import BaseModel
from typing import Optional


class GoogleBookVolumeInfo(BaseModel):
    title: str
    authors: Optional[list[str]] = None
    description: Optional[str] = None
    imageLinks: Optional[dict[str, str]] = None
    pageCount: Optional[int] = None
    publishedDate: Optional[str] = None
    categories: Optional[list[str]] = None


class GoogleBookItem(BaseModel):
    id: str
    volumeInfo: GoogleBookVolumeInfo


class GoogleBooksResponse(BaseModel):
    items: Optional[list[GoogleBookItem]] = None
    totalItems: int

File: frontend/src/types/google-books.ts
export interface GoogleBookItem {
  id: string
  volumeInfo: {
    title: string
    authors?: string[]
    description?: string
    imageLinks?: {
      thumbnail?: string
      smallThumbnail?: string
    }
    pageCount?: number
    publishedDate?: string
    categories?: string[]
  }
}

export interface GoogleBooksResponse {
  items?: GoogleBookItem[]
  totalItems: number
}
